source/daplink/interface/swd_host.c

```diff
@@ -170,7 +170,11 @@ uint8_t swd_read_ap(uint32_t adr, uint32_t *val)
     uint8_t tmp_in, ack;
     uint8_t tmp_out[4];
     uint32_t tmp;
+#ifdef MUSCA_A_BOARD
```

Comment:
Please add a clean method for configuring the required APSEL, as target-specific logic is not allowed in this common source file. I recommend adding an API to swd_host to set the current APSEL, a field to `target_family_descriptor_t`, and logic to `init_family()` that sets the APSEL based on the selected family.